C51-109: Styling display scrolling files tab

swastikpareek opened the pull request:
## Overview 
  This PR covers styling of the remaining File's Tab.

## Before
<img width="1680" alt="c51-109 - before" src="https://user-images.githubusercontent.com/3340537/46203076-40589e80-c336-11e8-8546-d37e64801f00.png">

## After 
 
<img width="1680" alt="c51-109 - after" src="https://user-images.githubusercontent.com/3340537/46203100-56665f00-c336-11e8-91d7-3a76551272b4.png">


## Technical Details
* Add a new dropdown template for file type tasks `~/civicase/ActivityCard--Menu--File.html`
* Add Download All feature to the action dropdown
* New BEM classes
  * civicase__file-filters-container - for filter containers
  * civicase__file-filters - for filter filters
  * civicase__file-tab-actions - for filter actions
  * civicase__file-tab-filters - file tab filters
* Moved `delete` scope function (For deleting file) from `FileList` directive to `Activity Card` as 
 `deleteFile` directive. This fixes delete file functionality on files tab and adds it to the activity card as a feature
   **Solution**: While moving the function from one directive to another we need to make sure that the correct refresh function is being called in the correct context in which `activity-card` is being loaded.
So for files tab, the context was `FileList` and the refresh function was `apiCtrl.refresh()` which is added a `refresh-callback` to the activity-card directive.
Now to the refresh function to be called for activity feed function was `refreshAll()` defined under `ActivityFeed.js` as the activity card was being called in the activity feed context on activity feed tab.
So in `refresh-callback` attribute of activity-card directive used under Activity feed was `refreshAll`.

* For CaseFIles tab to have bulk action abstract implementation, `intitiateBulkActions` fn is added in `civicaseCaseDetailsFileTab` directive (to check if `bulkAllowed `) and the same value is being passed to the `civicaseFileList` directive which is supplied to the activity card 

* Fix the sticky header issue 
  Sticky header initially bootstrapped on just loading but it should also consider the fact that it only loads when it is in view (i.e. case doesn't have focussed view). So in the directive this condition is added.
* The header used to show glitch in stickiness when the dropdown was opened. To fix it the old code reference from master was taken and the same logic is being implemented again
 reference [here](https://github.com/compucorp/uk.co.compucorp.civicase/blob/master/ang/civicase/Search.js#L38)
  ***Note**: Search.js doesn't call `affix` function so that check is also removed from the tests.*

* Also adds `$scope.areDetailsLoaded` var to check if the case details are ready to be showcased in the summary tab. This improves the UX as user will only view the data once it has been formatted properly on the UI. (Added the tests likewise)

* For counting overdue and scheduled tasks a new function is added inside the tests which will return the correct value of overdue and scheduled tasks as per the current date.

reneolivo commented on ang/civicase/CaseListDirectives.js:
```diff
@@ -32,51 +32,68 @@
 
       (function init () {
         scope.$watch('isLoading', checkIfLoadingCompleted);
+        scope.$watch('caseIsFocused', checkIfCaseIsFocused);
```
you only need one function:

```js
scope.$watch('isLoading', fixPositioningOnLoadingOrFocusChange);
scope.$watch('caseIsFocused', fixPositioningOnLoadingOrFocusChange);
```

```js
function fixPositioningOnLoadingOrFocusChange () {
  if (!scope.loading && !scope.caseIsFocused) {
    computeFixPositioning();
  }
}
```

reneolivo commented on ang/civicase/FileFilter.js:
```diff
@@ -1,31 +1,43 @@
+      templateUrl: '~/civicase/FileFilter.html',
+      controller: civicaseFileFilterController,
+      scope: {
+        apiCtrl: '=civicaseFileFilter'
```
why is this called  `apiCtrl`? `fileFilter` seems to be more adequate.

swastikpareek commented on ang/civicase/CaseDetailsFileTab.html:
```diff
@@ -1,13 +1,13 @@
-  <div class="panel-body clearfix" civicase-file-list="getfiles"></div>
+<div class="civicase__file-tab-actions">
+    <!--    For Bulk Actions
+   <div class="civicase__file-filter-actions"></div> -->
```
This will be used for implementing bulk actions. This div will contains bulk actions HTML. This has been placed here to give the next developer idea where he/she needs to place the HTML.

swastikpareek commented:
@reneolivo : Thanks for raising this. I have updated the PR with the your suggestions and also fixed the  js error that was coming up previously.
![no error logs](https://user-images.githubusercontent.com/3340537/46333706-fa098500-c63e-11e8-8cb6-58795dd8a6db.gif)
